### ISSUE TITLE
SCM view is blank in dev container in code 1.42

### ISSUE THREAD

**Issue by fsackur** [microsoft/vscode-remote-release#2185]
Following insiders update yesterday, the SCM (git) view is empty.

What I see:
![image](https://user-images.githubusercontent.com/3678789/72795522-b8d32500-3c35-11ea-83c3-ee242910a513.png)

When running insiders locally:
![image](https://user-images.githubusercontent.com/3678789/72795552-c688aa80-3c35-11ea-927b-fb71dad1b240.png)

- VSCode Version:
![image](https://user-images.githubusercontent.com/3678789/72794643-4281f300-3c34-11ea-92a9-2d6c0bc48cf7.png)

- Local OS Version: Win 10 1909
- Remote OS Version: Docker host: CentOS 7. Container image: ubuntu:18.04
- Remote Extension/Connection Type: Docker

Steps to Reproduce:

1. Connect to dev container
2. Open SCM pane

Does this issue occur when you try this locally?: No
Does this issue occur when you try this locally and all extensions are disabled?: Not tested, but issue still occurs when I rebuild the container with all remote extensions disabled

Potentially relevant: I mount .git-credentials and .gitconfig files directly in from host. This gives frequent `fatal: unable to write credential store: Device or resource busy` messages but has not been causing issues for the last month or two so I doubt it is the issue.

**Comment by fsackur** [comment]
```
[1 ms] Start: Resolving remote
[19 ms] Setting up container for folder or workspace: c:\Githubdata\mono-vscode

[98 ms] Start: Check Docker is running
[157 ms] Start: Starting container
[159 ms] Start: Run: docker run -a STDOUT -a STDERR --mount src=/home/<redacted_my_username>/mono,dst=/mono,type=bind -l vsch.quality=insider -l vsch.remote.devPort=0 -l vsch.local.folder=c:\Githubdata\mono-vscode -u <redacted_my_username> --network sl_<redacted_my_username> --add-host 1055033-BTS1.intensive.int:10.8.6.36 --add-host 1055034-BTS2.intensive.int:10.8.6.37 -v /var/run/docker.sock:/var/run/docker.sock -v /home/<redacted_my_username>/.ssh:/home/<redacted_my_username>/.ssh -v /home/<redacted_my_username>/mono:/home/<redacted_my_username>/mono -v /home/<redacted_my_username>/mono:/iawl -v /home/<redacted_my_username>/mono-vscode:/home/<redacted_my_username>/mono-vscode -v /home/<redacted_my_username>/mono-vscode/.devcontainer:/home/<redacted_my_username>/mono/.devcontainer -v /home/<redacted_my_username>/mono-vscode/.devcontainer:/mono/.devcontainer -v /home/<redacted_my_username>/mono-vscode/.vscode:/mono/.vscode -v /home/<redacted_my_username>/stepladder:/home/<redacted_my_username>/stepladder -v /home/<redacted_my_username>/stepladder_db_data:/home/<redacted_my_username>/stepladder_db_data -v /home/<redacted_my_username>/stepladder_redis_data:/home/<redacted_my_username>/stepladder_redis_data -v /home/<redacted_my_username>/stepladder_rabbitmq_data:/home/<redacted_my_username>/stepladder_rabbitmq_data -v /home/<redacted_my_username>/.git-credentials:/home/<redacted_my_username>/.git-credentials -v /home/<redacted_my_username>/.gitconfig:/home/<redacted_my_username>/.gitconfig -v /home/<redacted_my_username>/stepladder_secrets.sh:/home/<redacted_my_username>/stepladder_secrets.sh --entrypoint /bin/sh devcontainer:<redacted_my_username> -c echo Container started ; while sleep 1; do :; done
[723 ms] 
[3282 ms] Start: Inspecting container
[3287 ms] Container started

[3317 ms] Start: Run in container: uname -m
[4468 ms] x86_64
[4469 ms] 
[4469 ms] Start: Run in container: (cat /etc/os-release || cat /usr/lib/os-release) 2>/dev/null
[4492 ms] NAME="Ubuntu"
VERSION="18.04.3 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.3 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
[4492 ms] 
[4493 ms] Start: Run in container: cat /etc/passwd
[4500 ms] Start: Updating configuration state
[4502 ms] Start: Setup shutdown monitor
[4504 ms] Forking shutdown monitor: c:\Users\<redacted_my_username>\.vscode-insiders\extensions\ms-vscode-remote.remote-containers-0.97.0\dist\shutdownMonitorProcess \\.\pipe\vscode-remote-containers-8cccba6760d363052903be023d2178454a9f8e28-sock singleContainer Info c:\Users\<redacted_my_username>\AppData\Roaming\Code - Insiders\logs\20200121T094338\exthost1\ms-vscode-remote.remote-containers
[4522 ms] Start: Run in container: test -d /home/<redacted_my_username>/.vscode-server-insiders
[4540 ms] 
[4540 ms] 
[4540 ms] Exit code 1
[4540 ms] Start: Run in container: test -d /home/<redacted_my_username>/.vscode-remote
[4573 ms] 
[4574 ms] 
[4574 ms] Exit code 1
[4575 ms] Start: Run in container: set -o noclobber ; mkdir -p '/home/<redacted_my_username>/.vscode-server-insiders/data/Machine' && { > '/home/<redacted_my_username>/.vscode-server-insiders/data/Machine/.writeMachineSettingsMarker' ; } 2> /dev/null
[4588 ms] 
[4588 ms] 
[4588 ms] Start: Run in container: mkdir -p '/home/<redacted_my_username>/.vscode-server-insiders/data/Machine' && cat >'/home/<redacted_my_username>/.vscode-server-insiders/data/Machine/settings.json' <<'settingsJSON'
[4596 ms] 
[4596 ms] 
[4597 ms] Start: Run in container: test -d /home/<redacted_my_username>/.vscode-server-insiders/bin/d72b2d372f5024a9193b37a3830bbbc459c0ef1b
[4600 ms] 
[4601 ms] 
[4601 ms] Exit code 1
[4601 ms] Installing VS Code Server for commit d72b2d372f5024a9193b37a3830bbbc459c0ef1b
[4601 ms] Start: Run in container: mkdir -p /home/<redacted_my_username>/.vscode-server-insiders/bin/d72b2d372f5024a9193b37a3830bbbc459c0ef1b_1579601354503
[4614 ms] 
[4614 ms] 
[4616 ms] Start: Run: docker exec -w /home/<redacted_my_username>/.vscode-server-insiders/bin/d72b2d372f5024a9193b37a3830bbbc459c0ef1b_1579601354503 -u <redacted_my_username> -i 3e5eb9b5f57d2489cd41be35331ce7ca06634320e0276ef11244a8780ab722e8 tar --no-same-owner -xz --strip-components 1
[8995 ms] 
[8997 ms] 
[8997 ms] Start: Run in container: mv -n /home/<redacted_my_username>/.vscode-server-insiders/bin/d72b2d372f5024a9193b37a3830bbbc459c0ef1b_1579601354503 /home/<redacted_my_username>/.vscode-server-insiders/bin/d72b2d372f5024a9193b37a3830bbbc459c0ef1b
[9042 ms] 
[9042 ms] 
[9043 ms] Start: Launching Remote-Containers helper.
[9058 ms] Start: Run in container: # Copy .gitconfig
[9071 ms] .gitconfig exists
[9071 ms] 
[9071 ms] Start: Run in container: cat <<'EOF-/tmp/vscode-remote-containers-983b9d702f9f808e9755c31fe4c00405f3e3989d.js' >/tmp/vscode-remote-containers-983b9d702f9f808e9755c31fe4c00405f3e3989d.js
[9078 ms] 
[9078 ms] 
[9078 ms] Start: Run in container: set -o noclobber ; mkdir -p '/home/<redacted_my_username>/.vscode-server-insiders/data/Machine' && { > '/home/<redacted_my_username>/.vscode-server-insiders/data/Machine/.installExtensionsMarker' ; } 2> /dev/null
[9092 ms] 
[9092 ms] 
[9092 ms] Start: Run in container: cat <<'EOF-/tmp/vscode-remote-containers-server-983b9d702f9f808e9755c31fe4c00405f3e3989d.js' >/tmp/vscode-remote-containers-server-983b9d702f9f808e9755c31fe4c00405f3e3989d.js
[9101 ms] 
[9101 ms] 
[9101 ms] Start: Run in container: set -o noclobber ; mkdir -p '/home/<redacted_my_username>/.vscode-server-insiders/data/Machine' && { > '/home/<redacted_my_username>/.vscode-server-insiders/data/Machine/.postCreateCommandMarker' ; } 2> /dev/null
[9118 ms] 
[9119 ms] 
[9119 ms] Start: Run in container: test -f /home/<redacted_my_username>/.vscode-server-insiders/.postCreateCommandMarker
[9130 ms] 
[9131 ms] 
[9131 ms] Exit code 1
[9131 ms] Start: Run in container: for pid in `cd /proc && ls -d [0-9]*`; do { echo $pid ; readlink -f /proc/$pid/cwd ; xargs -0 < /proc/$pid/environ ; xargs -0 < /proc/$pid/cmdline ; } ; echo ; done 2>/dev/null
[9178 ms] Start: Run: docker exec -w /home/<redacted_my_username>/.vscode-server-insiders/bin/d72b2d372f5024a9193b37a3830bbbc459c0ef1b -u <redacted_my_username> -e SHELL=/usr/bin/zsh -e VSCODE_AGENT_FOLDER=/home/<redacted_my_username>/.vscode-server-insiders 3e5eb9b5f57d2489cd41be35331ce7ca06634320e0276ef11244a8780ab722e8 /home/<redacted_my_username>/.vscode-server-insiders/bin/d72b2d372f5024a9193b37a3830bbbc459c0ef1b/server.sh --disable-user-env-probe --port 0
[9689 ms] 
[11266 ms] 
[11305 ms] 
*
* Visual Studio Code Server
*
* Reminder: You may only use this software with Visual Studio family products,
* as described in the license https://aka.ms/vscode-remote/license
*




[11349 ms] IP Address: 10.0.1.152
IP Address: 172.18.0.11
Extension host agent listening on 33542


[10:09:21] Extension host agent started.

[11349 ms] Start: Run in container: echo 33542 >/home/<redacted_my_username>/.vscode-server-insiders/bin/d72b2d372f5024a9193b37a3830bbbc459c0ef1b/.devport
[11353 ms] 
[11353 ms] 
[11354 ms] Forwarding local port 33542 to container port 33542
[13351 ms] [10:09:23] [::ffff:127.0.0.1][6da845fd][ExtensionHostConnection] New connection established.
[10:09:23] [::ffff:127.0.0.1][6da845fd][ExtensionHostConnection] <144> Launched Extension Host Process.

[13472 ms] [10:09:23] [::ffff:127.0.0.1][c0d81d7d][ManagementConnection] New connection established.
```

**Comment by fsackur** [comment]
Output from Git:
```
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git ls-files --stage -- /mono/stepladder/stepladder/modules/iou/ticket_manager.py
> git cat-file -s 85028e19c8a8e54d0a79b11ca38f5c08b861426f
> git show :<redacted_path_in_repo>
> git status -z -u
> git symbolic-ref --short HEAD
> git rev-parse 4022-rc2
> git rev-parse --symbolic-full-name 4022-rc2@{u}
> git rev-list --left-right 4022-rc2...refs/remotes/origin/4022-rc2
> git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> git remote --verbose
> git config --get commit.template
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
```

**Comment by fsackur** [comment]
I include the above for completeness, none of the other output windows appear to show anything relevant (to my eyes). Let me know if there are any specific logs that would be helpful.

**Comment by chrmarti** [comment]
Does the Explorer viewlet show the expected files when connected to the container? And is this still a problem with the latest Insiders build? /cc @joaomoreno

**Comment by fsackur** [comment]
The explorer viewlet looks normal, all the other viewlets look normal.

Just tried with latest Insiders update, no change. Update was:
![image](https://user-images.githubusercontent.com/3678789/73188801-538ea080-411b-11ea-99d8-7c7614a38653.png)

**Comment by joaomoreno** [comment]
Are there any errors in the Devtools? `F1`, `Developer Tools`.

**Comment by fsackur** [comment]
```
log.ts:191  WARN Workbench did not finish loading in 10 seconds, that might be a problem that should be reported.
2console.ts:137 [Extension Host] (node:59808) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead. (at writeOut (internal/process/warning.js:27:3))
t.log @ console.ts:137
2log.ts:197   ERR read ECONNRESET: Error: read ECONNRESET
	at TCP.onStreamRead (internal/stream_base_commons.js:183:27)
```

**Comment by joaomoreno** [comment]
That's the entire output? When you open the SCM view and the view is empty, aren't there any more errors in the dev tools console?

**Comment by fsackur** [comment]
Ah, there are also these verbose messages which I didn't see at first:
```
event.ts:19 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
onFirstListenerAdd @ event.ts:19
_event._event @ event.ts:511
(anonymous) @ event.ts:56
onFirstListenerAdd @ event.ts:114
_event._event @ event.ts:511
N @ listWidget.ts:544
N @ abstractTree.ts:1083
createMouseController @ abstractTree.ts:1167
H @ listWidget.ts:1266
P @ abstractTree.ts:1163
t.AbstractTree @ abstractTree.ts:1291
r @ objectTree.ts:33
G @ listService.ts:742
_createInstance @ instantiationService.ts:117
createInstance @ instantiationService.ts:82
createSearchResultsView @ searchView.ts:707
renderBody @ searchView.ts:342
render @ paneview.ts:202
render @ viewPaneContainer.ts:137
onDidAddViews @ viewPaneContainer.ts:623
create @ viewPaneContainer.ts:358
create @ panecomposite.ts:42
showComposite @ compositePart.ts:216
doOpenComposite @ compositePart.ts:147
openComposite @ compositePart.ts:111
doOpenViewlet @ sidebarPart.ts:261
openViewlet @ sidebarPart.ts:212
run @ activitybarActions.ts:100
runAction @ actions.ts:198
run @ actions.ts:190
onClick @ actionbar.ts:180
(anonymous) @ actionbar.ts:150
processImmediate @ internal/timers.js:439
Show 3 more frames
[Violation] Forced reflow while executing JavaScript took 50ms
```
Other than that, nothing

**Comment by fsackur** [comment]
This is fixed now, as of 1.43.0 (17c6d12)
